query-coordinator/src/main/scala/com/socrata/querycoordinator/caching/cache/CacheSessionProvider.scala

```diff
@@ -1,7 +1,36 @@
 package com.socrata.querycoordinator.caching.cache
 
 import com.rojoma.simplearm.v2.ResourceScope
+import com.socrata.soql.SoQLAnalysis
+import com.socrata.soql.types.SoQLType
+import org.joda.time.DateTime
+import org.slf4j.Logger
 
 trait CacheSessionProvider {
-  def open(rs: ResourceScope): CacheSession
+  protected val log: Logger
+
+  def open(rs: ResourceScope, dataset: String): CacheSession
+
+  private var disabledTil = new DateTime(0)
+
+  def disabled() = {
+    disabledTil = new DateTime(System.currentTimeMillis()).plusHours(1)
+    log.warn(s"disable cache until $disabledTil" )
+  }
+
+  def isDisabled(): Boolean = {
```

Comment:
..and scala-style is to not use the "is" prefix and have no parens on this sort of thing.